framework/src/systems/SystemBase.C

```diff
@@ -1351,14 +1359,32 @@ SystemBase::name() const
   return system().name();
 }
 
+void
+SystemBase::initSolutionState()
+{
+  // Default is the current solution
+  unsigned int state = 0;
+
+  // Add additional states as required by the variable states requested
+  for (const auto & var : getVariables(/* tid = */ 0))
+    state = std::max(state, var->needSolutionState());
+  for (const auto & var : getScalarVariables(/* tid = */ 0))
+    state = std::max(state, var->needSolutionState());
+
+  needSolutionState(state);
+}
+
+std::string
+SystemBase::oldSolutionStateVectorName(const unsigned int state) const
+{
+  mooseAssert(state != 0, "Not an old state");
```

Comment:
I'm guessing this isn't called very often. If not, I'd make this a `mooseError`

Reply:
This is a private method. I would prefer the @rwcarlsen philosophy of using assertions for development-based error catching here

Reply:
K